### DISCUSSION TITLE
Is there a way to import a third party tailwind library without importing its styles explicitly?

### DISCUSSION THREAD

**Discussion by berkerdemirer** [tailwindlabs/tailwindcss#14137] [Help]
Hello,

I have created a react app that uses Tailwind and a vite-powered component library that uses Tailwind. I bundled the tailwind component library using rollup and made a bunch of js and d.ts files.

<img width="264" alt="image" src="https://github.com/user-attachments/assets/04f78472-0e36-4507-9200-c8554ae54de1">

Now, I would like to use the components inside the react app. In the main.tsx file of the react app, I imported the styles like this:

```
import '@mylib/webshop-components/style.css';
import '@/index.css';
```

My tailwind config looks like this in the react app: 

```
const config = {
  presets: [tailwindConfig],
  prefix: '',
  content: [
    './src/**/*.{js,ts,tsx}',
    './node_modules/@mylib/webshop-components/components/**/*.{js,ts,jsx,tsx,mdx}',
  ],
} satisfies Partial<Config>;

export default config;
```

Now, everything works like that but according to the docs, it should work only with the tailwind content config however, if I remove the style import of the component library from main.tsx file then everything breaks. It almost looks like pointing to node_modules make 0 difference in this scenario. 

It seems that the order of style imports also matters, if I import index.css first then things will look broken as well.

I already prefixed classes inside of the component library. 

Any ideas?

**Comment by wongjn** [comment]
> I already prefixed classes inside of the component library.

You have an empty `prefix` value in your Tailwind configuration, so Tailwind won't pick up any of the prefixed classes in your component library. Your `prefix` Tailwind configuration would override any `prefix` value defined in any `presets`.

**Reply by berkerdemirer** [reply]
But, what's the correct way to do this here? If I import two component libraries using Tailwind with two separate prefixes, how can I point them inside the react-app that uses Tailwind? 

libA prefix: la-
libB prefix: lb-
react-app prefix: none

What's the prefix config of the react-app here?

Because, when I remove the prefixes from those libraries then react-app tailwind config can use the correct config I guess. But, is this what we want? Then what's the purpose of prefixing?

@wongjn

**Reply by wongjn** [reply]
> Then what's the purpose of prefixing?

As per [the documentation](https://tailwindcss.com/docs/configuration#prefix):

> The `prefix` option allows you to add a custom prefix to all of Tailwind’s generated utility classes. This can be really useful when layering Tailwind on top of existing CSS where there might be naming conflicts.

---

> But, is this what we want? 

If the libraries' Tailwind configurations don't conflict with each other, you may not need to prefix the classes in the libraries. If their respective Tailwind configurations *do* conflict, then perhaps importing their *compiled* CSS may be an alternative solution.

**Reply by berkerdemirer** [reply]
Looks like I have to choose either option. If I use a prefix then I have to use the compiled CSS import, if not then I might alter the tailwind config file. It is hard to tell whether something will clash or not from the get-go and it would be problematic to add prefixes everywhere in the future.

Is there a standard way of doing this in the Tailwind world? What would be your personal preference?

**Reply by berkerdemirer** [reply]
@wongjn
